cellrank/tl/_utils.py

```diff
@@ -769,6 +769,16 @@ def _connected(c: Union[spmatrix, np.ndarray]) -> bool:
     return nx.is_connected(G)
 
 
+def _irreducible(d: Union[spmatrix, np.ndarray]) -> bool:
+    """Check whether the uirected graph encoded by d is irreducible."""
+
+    import networkx as nx
+
+    G = nx.DiGraph(d) if not isinstance(d, nx.DiGraph) else d
+
+    return len(list(nx.strongly_connected_components(G))) == 1
```

Comment:
Minor issue:
this will require to full evaluete the connected components (can be slow?). We just need to poll first 2 values to know
for sure whether it's irreducible or not. The function returns a generator, should be doable with `next` + `StopIteration` exception (I'd check whether it's a null graph as well first).

Reply:
Hi Mike, finding connected components runs in linear time and is thus very fast. If you have something in mind already then please feel free to add, otherwise, I would leave this as it is. What's a null graph?